bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/internal/Ism8Handler.java

```diff
@@ -0,0 +1,170 @@
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.ism8.server.DataPointChangedEvent;
+import org.openhab.binding.ism8.server.IDataPoint;
+import org.openhab.binding.ism8.server.IDataPointChangeListener;
+import org.openhab.binding.ism8.server.Server;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Ism8Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Hans-Reiner Hoffmann - Initial contribution
+ */
+@NonNullByDefault
+public class Ism8Handler extends BaseThingHandler implements IDataPointChangeListener {
+    private final Logger logger = LoggerFactory.getLogger(Ism8Handler.class);
+
+    private @Nullable Ism8Configuration config;
+    private @Nullable Server server;
+
+    public Ism8Handler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Ism8: Handle command = {} {}", channelUID.getId(), command);
+        if (isLinked(channelUID)) {
+            Channel channel = null;
+            for (Channel ch : getThing().getChannels()) {
+                if (ch.getUID().getId().equals(channelUID.getId())) {
+                    channel = ch;
+                    break;
+                }
+            }
+
+            if (channel != null && this.server != null) {
+                Server svr = this.server;
+                if (channel.getConfiguration().containsKey("id") && channel.getConfiguration().containsKey("write")
+                        && channel.getConfiguration().get("write").toString().equalsIgnoreCase("true")) {
+                    int id = Integer.parseInt(channel.getConfiguration().get("id").toString());
+                    logger.info("Channel '{}' writting into ID '{}'", channel.getUID().getId(), id);
+                    updateState(channelUID, new QuantityType<>(command.toString()));
+
+                    IDataPoint dataPoint = null;
+                    for (IDataPoint dp : svr.getDataPoints()) {
+                        if (dp.getId() == id) {
+                            dataPoint = dp;
+                            break;
+                        }
+                    }
+
+                    if (dataPoint != null) {
+                        try {
+                            svr.sendData(dataPoint.createWriteData(command));
+                        } catch (Exception e) {
+                            logger.error("Writting to ISM DataPoint '{}' failed. '{}'", dataPoint.getId(),
+                                    e.getMessage(), e);
```

Comment:
This log is to understand if something went wrong for sending data to the ISM8. To my understanding logging should give end users a hint in case something does not work correctly. A log entry would be generated in case the network connection to the ISM gets interrupted or something else. I decided to write the native message of the exception to understand the true reason of such occurence. I classified this as 'error' as an 'information' is for my understanding classifying as working correctly. Such exception did result in not sending to the ISM8 as expected.
What is the best practice in OpenHab to handle such cases?

Reply:
You could take a look at the coding guidelines. There are the different log levels explained and when to use one. https://www.openhab.org/docs/developer/guidelines.html#f-logging

It's totally right to log the exception's message. What I meant is that you're logging a complete stack trace. This is a bit unreasonable, if the network fails.

The log level `error` should only be used if something severe happend, like a bug in you code has been encountered.

When the network fails, `warn` or even `debug`  would be adequate.

Reply:
OK, I see. Most of the messages I categorized as error, but there is indeed no need to get the full stack. Probably in all cases the message is sufficient to understand the reason.